tests/test_view.py

```diff
@@ -111,190 +104,235 @@ def normalize(self, vec):
         return vec / np.linalg.norm(vec)
 
 
+@unittest.skipIf(GITHUB_ACTIONS, "GUI is not available in GitHub Actions")
+class ViewCommonCameraTC(ViewCameraTB, unittest.TestCase):
+    def test_value_get_set(self):
+        c = self.camera
+
+        for camera_type in ["fps", "orbit"]:
+            self.widget.setCameraType(camera_type)
+
+            c.linear_speed = 123.0
+            self.assertEqual(c.linear_speed, 123.0)
+
+            c.look_speed = 456.0
+            self.assertEqual(c.look_speed, 456.0)
+
+    def test_vector_get_set(self):
+        c = self.camera
+
+        for camera_type in ["fps", "orbit"]:
+            self.widget.setCameraType(camera_type)
+
+            c.position = (1, 2, 3)
+            c.view_center = (4, 5, 6)
+            c.up_vector = (7, 8, 9)
+
+            self.assertEqual(c.position, (1, 2, 3))
+            self.assertEqual(c.view_center, (4, 5, 6))
+            self.assertEqual(c.up_vector, (7, 8, 9))
+
+    def test_default_values(self):
+        c = self.camera
+
+        for camera_type in ["fps", "orbit"]:
+            self.widget.setCameraType(camera_type)
+
+            c.default_position = (1, 2, 3)
+            c.default_view_center = (4, 5, 6)
+            c.default_up_vector = (7, 8, 9)
+            c.default_linear_speed = 123.0
+            c.default_look_speed = 456.0
+
+            c.reset()
+
+            self.assertEqual(c.position, (1, 2, 3))
+            self.assertEqual(c.view_center, (4, 5, 6))
+            self.assertEqual(c.up_vector, (7, 8, 9))
+            self.assertEqual(c.linear_speed, 123.0)
+            self.assertEqual(c.look_speed, 456.0)
+
+
 @unittest.skipIf(GITHUB_ACTIONS, "GUI is not available in GitHub Actions")
 class ViewFPSCameraTC(ViewCameraTB, unittest.TestCase):
     camera_type = "fps"
```

Comment:
The testing code in `ViewFPSCameraTC` and `ViewOrbitCameraTC` looks very similar if not the same.  Is it possible to merge to a base class (`*TB`)?

Reply:
`test_rest`, `test_rotation` are actually different and `test_translation` indeed is the same in both fps and orbit. But it's not easy to move common tests to the base class, because unittest can't understand that structure.

I will move test_translation to `ViewCommonCameraTC`

Reply:
It's good to move identical testing code to a single place.  Thanks.

---

A side note.  For testing, it is not rare to duplicate code intentionally.  Compared to "feature" code, we more often change the testing code for experimenting without checking in the change (we want the testing code to be slightly more stable than the feature code).

In this PR, I do not see a reason to duplicate testing code.  This is just FYI.